### PR TITLE
feat: added testbed for v1

### PR DESCRIPTION
**Pull request by Dorfieeee**
# Description

Testbed for version 1.

We need to test our application for basic functionality before deployment.
Before we learn how to do it with [Jest](https://jestjs.io/), which is one of our next tasks after deployment,
it would be good to implement our own testing unit so we can imagine what it takes.

## How to test?
The first question to ask yourself could be: "How am I even going to test our app?". There are many
approaches, some worse, some better. Probably the worst approach in this case would be doing
it all manually. It is probably not always the worst approach, but we need to share our tests and have
a prove at hand.
My approach is to create a table of all possible cases for a user input and their expected results.
I then go through the table, apply an appropriate function on the input, compare its return value
with its expected result and display whether they match or not.
## What to test?
We want to make sure that when a user inserts a colour in any format(hex, hsl,...), it will be
either recognized or rejected for invalid input. When both colours are given and are recognized,
the next step is to calculate the contrast ratio.
The flow of our code is following: accept input -> convert input to RGB -> calculate constrast ratio -> display result
Therefore, it makes sense to start testing the conversion to RGB when the other functions depends on it.
If we're certain that any given input is correctly converted or rejected, we can test the calculation.
At the end, we can also test if the result of the calculation corresponds with our visual aid for different text
sizes.
1. test `getRGBfromColor()`
2. test `calculateRatio()`
3. visually test `handleTextStatus()`
## Adding test cases
`git pull <repository name> testbed-v1` branch to your local machine. Do not create another branch, make changes directly on the original.
When you open the `tests.js`, you should see two arrays, `RGBtestCases` and `contrastRatioTestCases`, with some
test cases already written. Add more test cases you think are necessary. Once you're done, simply `git push <repository name> testbed-v1` to update the branch with you cases.
## Running tests
Run the project `npm start`.
When `RUN_TESTS` in `tests.js` are set to `true`, you should be able to see the results of the tests in your browser's console on every page reload.

Happy bug hunting!

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore items (this includes basic clean up of files, or updates to packages)
- [ ] Updates to documentation

## Checklist

- [ ] I have a descriptive title for my PR
- [ ] I have linked this PR to the corresponding issue. [See how to do that here.](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] I have run the project locally and reviewed the code changes
- [ ] My changes generate no new warnings

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -110,7 +111,7 @@ const calculateRatio = (color1, color2) => {
     L2 is the relative luminance of the darker of the colors. */
   let lighterLum = Math.max(color1Luminance, color2Luminance);
   let darkerLum = Math.min(color1Luminance, color2Luminance);
-  return ((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2);
+  return Number(((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2));
```
Comment (Dorfieeee) [inline index.js:114]: I found a "bug" in `calculateRatio`. The function calculates a ratio but returned a string(`toFixed()` returns a string) which I think don't make much sense. It should return a number.
 I wrapped the whole expression with Number() constructor to ensure the returned value is a number.

File: index.js [hunk 2]
```diff
@@ -296,3 +297,10 @@ foregroundColor.onblur = displayResult;
 backgroundColor.onblur = displayResult;
 
 document.querySelector("#copyright").innerText = new Date().getFullYear();
+
+document.addEventListener("DOMContentLoaded", () => {
```
Comment (Dorfieeee) [inline index.js:301]: When everything is loaded and `RUN_TEST` is true, run tests.

File: index.js (hunk 1 above)
Comment (AndreyDmitriev39r) [reply index.js:114]: Makes sense, but this PR is about tests.
So it is out of the scope.

File: tests.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  ["HEX3", "#FFFFFF", true],
+  ["HEX3 invalid", "#FFX", false],
+  ["HEX3 no hash", "FFF", false],
+  ["HEX3 too short", "#FF", false],
+
+  // add more test cases
+];
+
+const contrastRatioTestCases = [
```
Comment (AndreyDmitriev39r) [inline tests.js:22]: if we are going to use it, more test cases required.

At least one for every possible format pair. The more - the better.

**Review by AndreyDmitriev39r — COMMENTED** [review]
Why do we need some tests like this, if we are going to use Jest in the near future?

Lets say we merge it now. What will we do with this code when Jest will come into play?

Why this PR exists without issue? The questions above - they should be answered/asked even before any kind of work is done, imo.

File: tests.js (hunk 3 above)
Comment (Dorfieeee) [reply tests.js:22]: When all inputs are converted to RGB format before calculation, it does not matter what format is given. They could all be in HEX.
Example:
Test case: `["hsl(240, 100%, 50%)", "#AAAB1C", 3.5]`, the two colours are first converted to RGB, then used in the calculation. So it does not matter if you write it like that or this `["rgb(33, 33, 222)", "rgb(170, 171, 28)", 3.5]`.

**Pull request closed without merging**